test/test_signature_inversion.py

```diff
@@ -0,0 +1,36 @@
+import torch
+
+from helpers import validation as v
+
+
+tests = ['invert_signature']
+depends = ['signature']
+signatory = v.validate_tests(tests, depends)
+
+
+
+def test_inverted_path_shape():
+    """Tests that the inverted path is of the right shape"""
+    for batch_size in (1, 2, 5):
+        for input_stream in (2, 3, 10):
+            for input_channels in (1, 2, 6):
+                path = torch.rand((batch_size, input_stream, input_channels))
+                for depth in (2, 4, 6):
+                    signature = signatory.signature(path, depth)
+                    inverted_path = signatory.invert_signature(signature, depth, input_channels, initial_position=path[:, 0, :])
+                    assert inverted_path.shape == (batch_size, depth + 1, input_channels)
```

Comment:
It would be nice to have some tests validating that the inversion produces a result close to the original path. Possibly by taking a short path length and high signature depth, to be able to do a good inversion?

If need be you can control the randomness by passing a seed to `torc.rand`, so that you can pick a seed that produces good results, rather than one that fails the inversion check every now and again.

Reply:
Yes I agree. I have added a test with the half circle example. Note that you have to set the tolerance very high (1e-01) to pass the test but I guess it is better than nothing.